src/main/java/webApp/webController.java

```diff
@@ -25,43 +30,120 @@ public String index(Model model){
     @PostMapping(value = "/createSurvey", consumes = "application/json", produces = "application/json")
     @ResponseBody
     public Response createSurvey(@RequestBody SurveyMessage surveyMessage) {
-        Survey survey = new Survey(surveyMessage.getName());
+
         Collection<Question> questionList = new ArrayList<>();
 
-        for (QuestionMessage question : surveyMessage.getQuestions()) {
-            if (question.getType().equals("openEnded")) {
-                questionList.add(new OpenEndedQuestion(question.getQuestion()));
-            } else if(question.getType().equals("numberQuestion")) {
-                questionList.add(new NumberQuestion(question.getQuestion(), question.getMin(), question.getMax()));
+        ArrayList<String> questionTexts = new ArrayList<>();
+
+        if (surveyMessage.getQuestions() == null) {
+            return new Response(null, "error", "Questions were null");
+        }
+
+        for (QuestionMessage questionMsg : surveyMessage.getQuestions()) {
+            if (questionMsg == null) {
+                return new Response(null, "error", "Question was null");
             }
+            questionTexts.add(questionMsg.getQuestion());
+            if (questionMsg.getType().equals(openQuestionType)) {
+                questionList.add(new OpenEndedQuestion(questionMsg.getQuestion()));
+            } else if(questionMsg.getType().equals(numQuestionType)) {
+                if (questionMsg.getMin() <= questionMsg.getMax()) {
+                    questionList.add(new NumberQuestion(questionMsg.getQuestion(), questionMsg.getMin(), questionMsg.getMax()));
+                } else {
+                    return new Response(null, "error",
+                            "Min is greater than max for question \"" + questionMsg.getQuestion() + "\"");
+                }
+            }
+        }
+
+        HashSet<String> set = new HashSet<>(questionTexts);
+
+        if(set.size() < questionTexts.size()){
+            return new Response(null, "error", "Duplicate questions detected");
```

Comment:
I don't think this is the best way of checking for duplicate questions, this may cause bugs in the near future.

Reply:
What is the alternative? What bugs?

Reply:
Maybe strings cause an issue if the HashSet checks for equality with = instead of `.equal`, but the tests cover that so if that were the case it should already be an error.